lib/cocoapods/validator.rb

```diff
@@ -319,6 +319,16 @@ def install_pod
       end.flatten
 
       @file_accessor = file_accessors.find { |accessor| accessor.spec.root.name == spec.root.name }
+
+      unless @file_accessor.nil?
+        dynamic_frameworks = @file_accessor.vendored_frameworks.select { |fw| `file #{fw + fw.basename('.framework')} 2>&1` =~ /dynamically linked/ }
+        dynamic_libraries = @file_accessor.vendored_libraries.select { |lib| `file #{lib} 2>&1` =~ /dynamically linked/ }
+        if (dynamic_frameworks.count > 0 || dynamic_libraries.count > 0) && consumer.platform_name == :ios &&
+            (deployment_target.nil? || Version.new(deployment_target).major < 8)
+          error('dynamic', 'Dynamic frameworks/libraries are only supported on iOS > 8.')
```

Comment:
Maybe this could be a bit more user friendly and use words instead of a more than operator.

> Dynamic frameworks and libraries and are only supported in iOS 8.0 and onwards.